tests/unit/algos/list/delimiterCurrent.spec.js

```diff
@@ -125,4 +129,253 @@ describe('DelimiterCurrent', () => {
 
         assert.deepStrictEqual(delimiter.result(), expectedResult);
     });
+
+    it('should return the object created before beforeDate', () => {
+        const beforeDate = '1970-01-01T00:00:00.003Z';
+        const delimiter = new DelimiterCurrent({ beforeDate }, fakeLogger, 'v1');
+
+        const masterKey1 = 'key1';
+        const date1 = '1970-01-01T00:00:00.004Z';
+        const value1 = `{"last-modified": "${date1}"}`;
+
+        assert.strictEqual(delimiter.filter({
+            key: makeV1Key(masterKey1),
+            value: value1,
+        }), FILTER_ACCEPT);
+
+        const masterKey2 = 'key2';
+        const date2 = '1970-01-01T00:00:00.000Z';
+        const value2 = `{"last-modified": "${date2}"}`;
+
+        assert.strictEqual(delimiter.filter({
+            key: makeV1Key(masterKey2),
+            value: value2,
+        }), FILTER_ACCEPT);
+
+        const expectedResult = {
+            Contents: [
+                {
+                    key: masterKey2,
+                    value: value2,
+                },
+            ],
+            IsTruncated: false,
+        };
+
+        assert.deepStrictEqual(delimiter.result(), expectedResult);
+    });
+
+    it('should return an empty list if last-modified is an empty string', () => {
+        const beforeDate = '1970-01-01T00:00:00.003Z';
+        const delimiter = new DelimiterCurrent({ beforeDate }, fakeLogger, 'v1');
+
+        const masterKey0 = 'key0';
+        const value0 = '{"last-modified": ""}';
+
+        assert.strictEqual(delimiter.filter({
+            key: makeV1Key(masterKey0),
+            value: value0,
+        }), FILTER_ACCEPT);
+
+        const expectedResult = {
+            Contents: [],
+            IsTruncated: false,
+        };
+
+        assert.deepStrictEqual(delimiter.result(), expectedResult);
+    });
+
+    it('should return an empty list if last-modified is undefined', () => {
+        const beforeDate = '1970-01-01T00:00:00.003Z';
+        const delimiter = new DelimiterCurrent({ beforeDate }, fakeLogger, 'v1');
+
+        const masterKey0 = 'key0';
+        const value0 = '';
```

Comment:
Let's make the value a valid JSON object to avoid triggering the "invalid JSON" condition:

```suggestion
        const value0 = '{}';
```